controllers/addon_types_test.go

```diff
@@ -12,17 +12,19 @@
  * limitations under the License.
  */
 
-package v1alpha1
+package controllers
```

Comment:
Why would this test move to controllers package and dir?

Reply:
Because the api_type tests depend on the fakeclient, which is generated from API
https://github.com/keikoproj/addon-manager/pull/116/files#diff-c099d0e1c963606f86c5036baa25836f1d77e9b5f2cf198dfe0a94398b528559R24

if this file is kept in the API directory, there is import loop.

Reply:
If this test is in the same package there's no need to import at all, it should be referenced as simply `Addon` correct?

Reply:
So client imports API is why?

Reply:
Thats a good idea, already covered under controllers/suite_test.go, just simply re-use.

Reply:
The controllers/suite_test.go is different usage, you have a simple api test here need to move under `/api/*` directory in a way that doesn't introduce diamond  dependency. Suite test from controllers package should not be re-used.

Reply:
717dd1fe56e813e78c27b84b151be93a6b0f8a16